app/scripts/controllers/detect-tokens.js

```diff
@@ -43,12 +46,18 @@ export default class DetectTokensController {
     tokensController,
     assetsContractController = null,
     trackMetaMetricsEvent,
+    getNetworkClientById,
+    disableLegacyInterval = false,
   } = {}) {
+    super();
+    this.getNetworkClientById = getNetworkClientById;
     this.messenger = messenger;
     this.assetsContractController = assetsContractController;
     this.tokensController = tokensController;
     this.preferences = preferences;
-    this.interval = interval;
+    if (!disableLegacyInterval) {
+      this.interval = interval;
+    }
```

Comment:
this was used for testing to not kick off a `setInterval` that happens in the setter for `this.interval`